src/ipc/transport/mmap.rs

```diff
@@ -81,6 +81,9 @@ impl Socket {
         }
     }
     pub fn send(&mut self, packet: &[u8]) -> Result<(), ConnectionError> {
+        if packet.len() > BUFFER_SIZE - 8 {
```

Comment:
`Bincode` will emit the length of `packet` to bytestream so `BUFFER_SIZE - 8` is not a strict upperbound. Let's check it when serializing.